metrics-tracing-context/src/lib.rs

```diff
@@ -125,6 +125,14 @@ impl TracingContextLayer<label_filter::IncludeAll> {
     }
 }
 
+impl TracingContextLayer<label_filter::Allowlist> {
+    /// Creates a new [`TracingContextLayer`] with an allowed list of label
+    /// names.
+    pub fn allowlist(label_names: &[&str]) -> Self {
```

Comment:
A naming and doc comment tweak to make it match the suggestions above:

```suggestion
    /// Creates a new [`TracingContextLayer`] that only allows labels contained
    /// in a predefined list.
    pub fn only_allow<I, S>(allowed: I) -> Self
    where
        I: IntoIterator<Item = S>,
        S: AsRef<str>,
    {
```